types/message.go

```diff
@@ -84,14 +98,12 @@ func (msg *Message) String() string {
 	return fmt.Sprintf("Message cid=[%v]: %s", cid, string(js))
 }
 
-// NewMessage creates a new message.
-func NewMessage(from, to address.Address, nonce uint64, value *AttoFIL, method string, params []byte) *Message {
-	return &Message{
-		From:   from,
-		To:     to,
-		Nonce:  Uint64(nonce),
-		Value:  value,
-		Method: method,
-		Params: params,
-	}
+// Equals tests whether two metered messages are equal
+func (msg *Message) Equals(other *Message) bool {
+	return msg.To == other.To &&
```

Comment:
trading off this way instead of using cid does make me nervous but i'm willing to try it